### ISSUE TITLE
Can't parse some UCSC hub files

### ISSUE THREAD

**Issue by NullModel** [GMOD/jbrowse-components#3461]
https://hgdownload.soe.ucsc.edu/hubs/GCF/000/002/985/GCF_000002985.6/hub.txt

trying to open that hub as a connection gives an error about blank lines

**Comment by NullModel** [comment]
That assembly hub can also be referenced as a track hub without the genome definitions:

  https://hgdownload.soe.ucsc.edu/hubs/GCF/000/002/985/GCF_000002985.6/trackDb.txt

**Comment by NullModel** [comment]
Also note, UCSC hosts alias files for these assembly hubs, for example:
  https://hgdownload.soe.ucsc.edu/hubs/GCF/000/002/985/GCF_000002985.6/GCF_000002985.6.chromAlias.txt

Also in a bigBed format for those cases of very large sequence count assemblies so names can
be retrived via the bigBed name index instead of trying to load such a large ascii file.

The .as schema in the file will define the columns, for example:

bigBedInfo -as GCF_002204515.2_AaegL5.0.chromAlias.bb

table chromAlias
   "chromAlias bigBed index"
    (
    string chrom;       "native sequence name"
    uint chromStart;    "always 0"
    uint chromEnd;      "chromosome size"
    string refseq;      "RefSeq name"
    string assembly;    "Assembly name"
    string genbank;     "GenBank name"
    string ncbi;        "NCBI name"
    string ucsc;        "UCSC name"
    )

The naming authority can vary depending upon what is available for an assembly.
This schema will define the naming authority.

See also:

  https://genomewiki.ucsc.edu/index.php?title=Chrom_Alias

**Comment by cmdcolin** [comment]
just for ref, I think this file (https://hgdownload.soe.ucsc.edu/hubs/GCF/000/002/985/GCF_000002985.6/hub.txt) is a so-called "single file hub", which we dont support yet...feature request for it here though https://github.com/GMOD/ucsc-hub-js/issues/17 (currently expect separate files for hub, genomes, trackDb) and we also don't yet at least allow loading the trackDb without the hub. all good possibilities to add here though! was great meeting you at the jb session

**Comment by parteekcoder** [comment]
@rbuels @cmdcolin hi can you please assign me this issue, I would like to contribute in your organization in GSOC this year

**Comment by rbuels** [comment]
@parteekcoder it's too early to assign specific issues as part of gsoc, you need to submit a gsoc application etc. If this issue is still open by the time the gsoc application process is finished, I'm sure you could work on it though

**Comment by parteekcoder** [comment]
I would like to contribute now so that I can understand your organization
better , And it will be helpful for me when I apply for GSOC this year in
your organization

**Comment by rbuels** [comment]
Sure, we would be very thankful for a contribution!  First step would be to make sure that the @gmod/ucsc-hub module (which is the parsing library we use) can parse this file correctly, submitting a PR on that repository (https://github.com/GMOD/ucsc-hub-js) if necessary, and after that you would want to troubleshoot why the hub isn't opening correctly in JBrowse itself

**Comment by parteekcoder** [comment]
Is GMOD participating GSOC this year ,

And you will assign this issue to me or I just do PR

Is there any community channel for conversation

**Comment by rbuels** [comment]
Yes, I think we will be participating in GSOC this year. I won't assign the issue, just do the PR. Community channel is on gitter: https://gitter.im/GMOD/jbrowse2

**Comment by parteekcoder** [comment]
I am not able to find GMOD project ideas for GSOC this year , can you
please share the link

**Comment by MadsaraManzar** [comment]
I am a beginner. I want to contribute in open source projects. Could anyone help me?

**Comment by cmdcolin** [comment]
@MadsaraManzar we are happy to help, but you also have to make an effort to explain to us why you want to contribute to our project in particular, what your skill set is, etcetera. 

it takes time and effort for us to invest explain this to you, and we already have our README doc. if that doesn't help you have to explain what issues you had.  in any case, this thread is getting off topic, i am closing it. further work will happen at https://github.com/GMOD/ucsc-hub-js/issues/17

**Comment by cmdcolin** [comment]
feel free to make new threads in for new topics https://github.com/GMOD/jbrowse-components/discussions

**Comment by rbuels** [comment]
reopening, this is still an issue. not sure why it was closed in the first place.

**Comment by rbuels** [comment]
@NullModel I'm taking a look at this now, and I'm noticing in the file linked above, there are some subtracks that are indented with 8 spaces, and some with 4. Is there any semantic difference between the two indentation levels? I can't detect a difference in the way the UCSC browser displays their subtrack entries.

![image](https://user-images.githubusercontent.com/73749/229635652-81c26cff-a097-484f-bced-fad48a3d5948.png)
![image](https://user-images.githubusercontent.com/73749/229635725-64db3ea5-404a-4f69-af53-a1802a699385.png)

**Comment by cmdcolin** [comment]
will update if we have url param or something like that
